client/lib/protect-form/index.tsx

```diff
@@ -143,3 +143,27 @@ export const checkFormHandler: PageJS.Callback = ( context, next ) => {
 		}, 0 );
 	}
 };
+
+type ProtectForm = {
+	markChanged: () => void;
+	markSaved: () => void;
+};
+
+export const useProtectForm = ( id: string ): ProtectForm => {
+	const _markSaved = React.useCallback( () => markSaved( id ), [ id ] );
+	const _markChanged = React.useCallback( () => markChanged( id ), [ id ] );
```

Comment:
Is it worth the cost to use `React.useCallback` here?

Reply:
This is quite useful especially for the `protectForm` HOC. When used as `protectForm( connect( ... )( Component ) )`, it doesn't defeat the `connect`-ed component's `shouldComponentUpdate` optimization by passing different props each time.